presto-main/src/main/java/com/facebook/presto/operator/scalar/RowToJsonCast.java

```diff
@@ -102,4 +106,32 @@ public static Slice toJson(List<JsonGeneratorWriter> fieldWriters, SqlFunctionPr
             throw new RuntimeException(e);
         }
     }
+
+    @UsedByGeneratedCode
+    public static Slice toJson(List<String> fieldNames, List<JsonGeneratorWriter> fieldWriters, SqlFunctionProperties session, Block block)
+    {
+        if (session.isFieldNamesInJsonCastEnabled()) {
+            return toJsonObject(fieldNames, fieldWriters, session, block);
+        }
+        return toJsonArray(fieldWriters, session, block);
+    }
+
+    private static Slice toJsonObject(List<String> fieldNames, List<JsonGeneratorWriter> fieldWriters, SqlFunctionProperties session, Block block)
+    {
+        try {
+            SliceOutput output = new DynamicSliceOutput(40);
```

Comment:
Need some explanation of the '40'.

Reply:
I cannot find any reasoning for why 40 was chosen initially but my guess is that this is a good proxy for the average size of the field value. Given that we will now be adding the fieldname to the output as well, I think I should make the estimated size of the slice 50(40 chars for value + 10 chars for the key). What do you think?

 
PR which introduced the change for context: https://github.com/prestodb/presto/pull/7565/files